### DISCUSSION TITLE
How to stop revalidating after data meets condition?

### DISCUSSION THREAD

**Discussion by jonasmerlin** [vercel/swr#792] [Q&A] [answered]
So I'm using SWR to query an endpoint with `refreshInterval` set to `1000`. But I only want to do this until the data returned meets a certain condition. After this, I want to set the refreshInterval to 0 again. What would be the best solution for this?

I guess I can see a solution with useRef for the `options` object for example, but that seems kind of overkill ...

**Comment by shuding** [comment, marked as answer]
I'd still use a state for it. It's not overkill because it **is** an app state (a ref will not trigger a re-render so a state should be better).

**Reply by jonasmerlin** [reply]
I see! Makes sense, thanks for clearing this up. :)
